openmc/deplete/results_list.py

```diff
@@ -296,3 +298,72 @@ def get_step_where(
             "relative tolerances {} and {}.".format(
                 time, time_units, atol, rtol)
         )
+
+    def export_to_materials(self, burnup_index, nuc_with_data=None):
+        """Return openmc.Materials created from these depletion results.
+
+        Parameters
+        ----------
+        burn_index : int
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            Nuclides iterable to evaluate with neutron data.
+            This must be specified if not all nuclides appearing in
+            depletion results have associated neutron cross sections, and
+            as such cannot be used in subsequent transport calculations.
+            If not provided, nuclides from the cross_sections element of
+            materials.xml will be used. If that element is not present,
+            nuclides from OPENMC_CROSS_SECTIONS will be used.
+
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            if not all(isinstance(nuclide_name, str) for nuclide_name in nuc_with_data):
+                raise InvalidArgumentError("Expected an iterable of strings as acceptable nuclide data.")
```

Comment:
This line exceeds 100 characters, which goes over my loose limit of 90ish.

Reply:
Also, I agree with @drewejohnson's original suggestion to use `check_iterable_type`

Reply:
Yeah, OK, I give, that way is a lot cleaner. Change is in now.